packages/blade/src/components/ActionList/ActionList.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import Box from '~components/Box';
+import { useDropdown } from '~components/Dropdown/useDropdown';
+import { Text } from '~components/Typography';
+import { getPlatformType, makeAccessible } from '~utils';
+
+type ActionListItemProps = {
+  title: string;
+  value: string;
+  href?: string;
+  /**
+   * Internally passed from ActionList. No need to pass it explicitly
+   */
+  index?: number;
+};
+const ActionListItem = (props: ActionListItemProps): JSX.Element => {
+  const {
+    activeIndex,
+    dropdownBaseId,
+    onOptionClick,
+    selectedIndices,
+    setShouldIgnoreBlur,
+    selectInputRef,
+  } = useDropdown();
+
+  const platformType = getPlatformType();
+  const renderOnWebAs = props.href ? 'a' : 'button';
+  const isReactNative = platformType === 'react-native';
+
+  return (
+    <Box
+      as={!isReactNative ? renderOnWebAs : undefined}
+      id={`${dropdownBaseId}-${props.index}`}
+      role="option"
+      data-value={props.value}
+      data-index={props.index}
+      aria-selected={
+        typeof props.index === 'number' ? selectedIndices.includes(props.index) : undefined
+      }
+      onClick={(e: React.MouseEvent<HTMLButtonElement>) => {
+        if (typeof props.index === 'number') {
+          onOptionClick(e, props.index);
+        }
+      }}
+      onFocus={() => {
+        // We don't want to keep the browser's focus on option item. We move it to selectInput
+        selectInputRef.current?.focus();
+      }}
+      onMouseDown={() => {
+        setShouldIgnoreBlur(true);
+      }}
+      href={props.href}
+      style={{
+        border: activeIndex === props.index ? '2px solid red' : '',
+        width: '100%',
+      }}
+    >
+      <Text>{props.title}</Text>
+    </Box>
+  );
+};
+
+type ActionListProps = {
+  children: React.ReactNode[];
+};
+const ActionList = ({ children }: ActionListProps): JSX.Element => {
```

Comment:
put the usage in comments as you have done in Dropdown. really liked it. gives the code reader an immediate visual usage 👌

Reply:
I have a separate documentation PR for adding storybook etc so I plan to add jsdoc in that PR.